### PR TITLE
GetSchema: Batch/parallel access to underlying mysqld for lower latency

### PR DESCRIPTION
**Pull request by teejae**
- One-pass access for PrimaryKeyColumns
  - `getPrimaryKeyColumns` retrieves multiple tables from `information_schema.key_column_usage` instead of using `SHOW INDEX`
- Parallel requests for Column metadata and Create Table schema
- Filters out tables before doing expensive operations

Signed-off-by: Toliver Jue <toliver@planetscale.com>

### CODE REVIEW

**Comment by teejae** [comment]
FYI, depends on PR https://github.com/vitessio/vitess/pull/6243 and https://github.com/vitessio/vitess/pull/6242.

Diff currently includes rebase from https://github.com/vitessio/vitess/pull/6242.

File: go/vt/mysqlctl/schema.go [hunk 1]
```diff
@@ -46,6 +46,15 @@ func (mysqld *Mysqld) executeSchemaCommands(sql string) error {
 	return mysqld.executeMysqlScript(params, strings.NewReader(sql))
 }
 
+// tableList returns an IN clause "('t1', 't2'...) for a list of tables."
+func tableListSql(tables []string) string {
+	if len(tables) == 0 {
+		return "()"
+	}
+
+	return "('" + strings.Join(tables, "', '") + "')"
```
Comment (systay) [inline go/vt/mysqlctl/schema.go:55]: Is this necessary?

**Comment by teejae** [comment]
FYI converted to draft to clean up PR.

**Comment by deepthi** [comment]
Merged upstream.

**Pull request closed without merging**